.github/workflows/robot-framework.yml

```diff
@@ -0,0 +1,51 @@
+name: Robot Framework CI
+
+# Borrowed code from:
+# https://piyathida-sanaoun01.medium.com/how-to-setup-ci-github-action-to-run-robot-framework-49028a404fee
+# https://github.com/laojala/robot_docker_demo/blob/master/.github/workflows/run_robot_framework_tests.yml
+
+on:
+  push:
+    branches:
+      - main
+    # Do not run tests if file README.md changes
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    # Do not run tests if file README.md changes
+    paths-ignore:
+      - '**.md'
+  schedule:
+    # Run tests nightly at 2am
+    - cron:  '0 2 * * *'
+jobs:
+    Run-Test:
+      runs-on: ubuntu-20.04
+      steps:
+        - name: Checkout repository
+          uses: actions/checkout@v1
+        - name: Install Google Chrome
+          run: |
+              wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+              sudo apt install ./google-chrome-stable_current_amd64.deb
+        # - uses: browser-actions/setup-chrome@v1
+        # test
+        - uses: nanasess/setup-chromedriver@v2
+        - uses: actions/cache@v2
+          id: cache
+          with:
+            path: ~/.cache/pip
+            key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements*.txt') }}
+        - name: Install pip dependencies
+          run: |
+            pip install --upgrade pip certifi urllib3
+            pip install -r requirements.txt -r requirements-test.txt
+        - name: Run Streamlit app
+          run: streamlit run streamlit_app.py &
+        - name: Run Robot tests
+          run: ./run-robot-tests
```

Comment:
You could potentially just add `streamlit run streamlit_app.py &` to this script itself, which would also be nice for local testing